c7n/resources/apigw.py

```diff
@@ -405,3 +405,165 @@ def process(self, resources):
                     resourceId=m['resourceId'],
                     httpMethod=m['httpMethod'],
                     patchOperations=ops)
+
+
+@RestResource.filter_registry.register('rest-integration')
+class FilterRestIntegration(ValueFilter):
+    """Filter rest resources based on a key value for the method integration of the resource
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: integrations-with-type-aws
+            resource: rest-resource
+            filters:
+              - type: rest-integration
+                key: type
+                value: AWS
+    """
+
+    schema = utils.type_schema(
+        'rest-integration',
+        method={'type': 'string', 'enum': [
+            'all', 'ANY', 'PUT', 'GET', "POST",
+            "DELETE", "OPTIONS", "HEAD", "PATCH"]},
+        rinherit=ValueFilter.schema)
+    permissions = ('apigateway:GET',)
+
+    def process(self, resources, event=None):
+
+        method_set = self.data.get('method', 'all')
+        # 10 req/s with burst to 40
+        client = utils.local_session(
+            self.manager.session_factory).client('apigateway')
+
+        futures = {}
+        results = []
+
+        with self.executor_factory(max_workers=2) as w:
+            tasks = []
+            for r in resources:
+                r_method_set = method_set
+                if method_set == 'all':
+                    r_method_set = r.get('resourceMethods', {}).keys()
+
+                for m in r_method_set:
+                    tasks.append((r, m))
+
+            for task_set in utils.chunks(tasks, 20):
+                futures[w.submit(
+                    self.process_task_set, client, task_set)] = task_set
+
+            for f in as_completed(futures):
+                task_set = futures[f]
+                for m in f.result():
+                    if self.match(m):
+                        results.append(m)
+
+        return results
+
+    def process_task_set(self, client, task_set):
+        results = []
+        for r, m in task_set:
+            try:
+                integration = client.get_integration(
+                    restApiId=r['restApiId'],
+                    resourceId=r['id'],
+                    httpMethod=m)
+            except client.exceptions.NotFoundException:
+                continue
+
+            integration.pop('ResponseMetadata', None)
+            integration['restApiId'] = r['restApiId']
```

Comment:
I understand this implementation is copy/pasted from the rest-method impl, but the preference would be to have these annotations on the method flyweights/resources with the `c7n:` prefix

Reply:
Think I got the output as it should be in my latest commit but welcome to suggestions if it's not to your liking.